### PR TITLE
Update OpenShift Kubernetes documentation

### LINKED ISSUE #3003: K8+OpenShift errors and inconsistencies
### How can we make our docs better?

What do you need?
-----------------

The document says our kubernetes integration supports Currently tested with versions 3.7, 3.9, 4.2 and 4.3.\
<https://docs.newrelic.com/docs/integrations/kubernetes-integration/get-started/kubernetes-integration-compatibility-requirements>

However, some components are not working with OpenShift 4.2 and 4.3. I found three things we should improve.

### Overall

We use `oc` command instead of `kubectl`. We need a note to notify for OpenShift users.

### APM Link (metadata-injection)

The document says we have to modify `master-config.yaml` for OpenShift 3.9 or **higher** to use the admission webhook.

<https://docs.newrelic.com/docs/integrations/kubernetes-integration/link-your-applications/link-your-applications-kubernetes#openshift-req>

However, OpenShift 4 no longer supports master-config.yaml. Red Hat says we should use [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) instead of editting master-config. Here is the Red Hat document available for free Red Hat Developer Program users.\
<https://access.redhat.com/solutions/5041951>

> **Issue**
>
> Is admission webhook supported in OpenShift 4?
>
> **Resolution**
>
> Yes it's supported since OpenShift 4.1 as described in the release notes.\
> Please refer to the Kubernetes Dynamic Admission Control document for details.

Could you add to the document how we can enable admission webhook with dynamic admission control since OpenShift 4.2 and 4.3 are supported?

### Logs

The steps described in our document is not working with OpenShift.\
<https://docs.newrelic.com/docs/logs/enable-logs/enable-logs/kubernetes-plugin-logs>

This is because our kubernetes plugin depends on `hostPath` mount feature.\
<https://github.com/newrelic/kubernetes-logging/blob/master/new-relic-fluent-plugin.yml#L70>

However, OpenShift doesn't support hostPath by default. We have to use the manual step (not recommended by the product team by required) and add some custom steps as follows.

Create configMap and RBAC objects before creating daemonset.

```
oc apply -f fluent-conf.yml
oc apply -f rbac.yml
```

Then configure the policy.

```
oc adm policy add-scc-to-user privileged system:serviceaccount:default:newrelic-logging
oc adm policy add-cluster-role-to-user cluster-reader
```

Finally, create the daemonset.

```
oc apply -f new-relic-fluent-plugin.yml
```

### Anything else you'd like to share?

Migrated from Jira DOC-5758

### PR DESCRIPTION
**Pull request by pizzaz93**
Fixes #3003

### CODE REVIEW

File: src/content/docs/logs/forward-logs/kubernetes-plugin-log-forwarding.mdx
Comment (kang-makes) [inline src/content/docs/logs/forward-logs/kubernetes-plugin-log-forwarding.mdx:62]: Based on what I said about the helm charts, this service account (`default`) seems incorrect.

If you use `nri-bundle`/guided install you may install it in the `newrelic` namespace instead of `newrelic-logging`. The helm chart [creates a daemonset](https://github.com/newrelic/helm-charts/blob/master/charts/newrelic-logging/templates/daemonset.yaml#L34) that [uses its own service account](https://github.com/newrelic/helm-charts/blob/master/charts/newrelic-logging/templates/serviceaccount.yaml) so `default` does not seem to be the service account that the logging daemonSet is going to use.

Not at least without understanding/explaining what the manifests described above install.

**Review by kang-makes — COMMENTED** [review]
I think this is a very necessary doc update. In fact, we have an issue reminding us that we have to revisit these docs about our integration: newrelic/nri-kubernetes#405

But I would wait for the logging team to re-review this PR in case I have missed something.

Thank you very much for bringing this to our attention.

File: src/content/docs/logs/forward-logs/kubernetes-plugin-log-forwarding.mdx
Comment (kang-makes) [reply src/content/docs/logs/forward-logs/kubernetes-plugin-log-forwarding.mdx:48]: @noly changed yesterday CODEOWNERS to all the logging team: @newrelic/logging

Comment (noly) [reply src/content/docs/logs/forward-logs/kubernetes-plugin-log-forwarding.mdx:48]: Yes, I did!! 😸

**Comment by x8a** [comment]
After talking to SMEs, we need to dismiss the changes in `/docs/logs/forward-logs/kubernetes-plugin-log-forwarding.mdx`. They're deprecating static manifests, so we don't want to add those steps for openshift. They're currently trying to see if helm charts work with openshift. 

Thanks again for the edits!

**Review by x8a — DISMISSED** [review]

**Review by x8a — APPROVED** [review]

**Pull request merged**
